jobs/FindLabeledImagesInResults.php

```diff
@@ -0,0 +1,172 @@
+            'description="' . $this->db->real_escape_string( $this->description ). '"'
+        );
+        return $this->db->insert_id;
+    }
+
+    private function getSearchTerms() : array {
+        $searchTerms = [];
+        $searchTermResults = $this->db->query(
+            'select distinct(term) as term from results_by_component where rating is not null'
+        );
+        while ( $searchTerm = $searchTermResults->fetch_assoc() ) {
+            $searchTerms[] = $searchTerm['term'];
+        }
+        return $searchTerms;
+    }
+
+    private function search( string $searchTerm ) : array {
+        curl_setopt( $this->ch, CURLOPT_URL, $this->getSearchUrl( $searchTerm ) );
+        $result = curl_exec( $this->ch );
+        if ( curl_errno( $this->ch ) ) {
+            $this->log( curl_error( $this->ch ) . ':' . curl_errno( $this->ch ) );
+            die( 'Exiting because of curl error, see log for details.' );
+        }
+        $array = json_decode( $result, true );
+        return $array;
+    }
+
+    private function getSearchUrl( string $searchTerm ) : string {
+        return sprintf(
+            $this->searchUrl . '/w/index.php?search=%s+filetype:bitmap&ns6=1&' .
+            'cirrusDumpResult&mediasearch=1&limit=100',
+            urlencode( $searchTerm )
+        );
+    }
+
+    private function processResults( string $searchTerm, array $searchResults, int $searchId ) {
+        $labeledData = $this->getLabeledData( $searchTerm );
+
+        $titles = [];
+        if ( isset( $searchResults['__main__']['result']['hits']['hits'] ) ) {
+            foreach ( $searchResults['__main__']['result']['hits']['hits'] as $index => $result ) {
+                $titles[] = $this->extractTitle( $result['_source'] );
+            }
+        }
+        $this->log( 'Found ' . count( $titles ) . ' results' );
+        if ( count( $titles ) > 0) {
+            $this->db->query(
+            'insert into resultset set ' .
+                'searchId=' . intval( $searchId ) . ', ' .
+                'term="' .  $this->db->real_escape_string( $searchTerm ) . '", ' .
+                'resultCount=' . intval( count( $titles ) )
+            );
+            $resultsetId = $this->db->insert_id;
+            $labeledImageCount = 0;
+            foreach ( $titles as $index => $title ) {
+                if ( isset( $labeledData[$title] ) ) {
+                    $this->db->query(
+                        'insert into labeledResult set ' .
+                        'resultsetId=' . intval( $resultsetId ) . ', ' .
+                        'filePage="' .  $this->db->real_escape_string( $title ). '", ' .
+                        'position=' . intval( $index ) . ', ' .
+                        'rating=' . intval( $labeledData[$title] )
+                    );
+                    $labeledImageCount++;
+                }
+            }
+            $this->log( $labeledImageCount . ' of the results were labeled' );
+        } else {
```

Comment:
Else branch seems unnecessary - the resultset insert will be the same anyway (since count() === 0), and the labeledResults inserts won't happen (won't loop over an empty array)